src/basic/terminal-util.c

```diff
@@ -857,6 +857,39 @@ unsigned lines(void) {
         return cached_lines;
 }
 
+int terminal_set_size_fd(int fd, unsigned rows, unsigned cols) {
+        struct winsize ws;
+
+        if (rows == UINT_MAX && cols == UINT_MAX)
+                return 0;
+
+        if (ioctl(fd, TIOCGWINSZ, &ws) < 0)
+                return log_debug_errno(errno, "TIOCSWINSZ ioctl for setting TTY size failed, ignoring: %m");
```

Comment:
In the message: TIOCSWINSZ → TIOCGWINSZ

"ignoring" → "not setting terminal size"? 

How likely is this or the other commands below to fail, e.g. with selinux? If the message is meant to be used really, not just a placeholder, it would be nice to provide the path. Would it be possible to make the function take an additional `const char* ident` parameter that is used in the message here?

Also, is it actually necessary to call this in most cases? If both rows and cols are set, then TIOCSWINSZ below can be used without TIOCGWINSZ.

Reply:
> Also, is it actually necessary to call this in most cases? If both rows and cols are set, then TIOCSWINSZ below can be used without TIOCGWINSZ.

Lennart mentioned that setting the tty size triggers the SIGWINCH signal on listening processes, so it's better to only do TIOCSWINSZ if the size we're going to configure is actually different from the size that's already configured. To accomplish that we always have to do TIOCGWINSZ so we know the current size.

Reply:
`struct winsize` has two additional fields besides col/row. They are documented as "unused", but I#d really stay away from them, i.e. never reset them but only ever hand back to the kernel what we got from it in the first place. The kernel copies the whole structure around after all, not just the two relevant fields.

Reply:
Apparently these fields aren#t actually unused, VTE sets them:

https://gitlab.gnome.org/GNOME/vte/-/issues/251

So I figure we should definitely make sure that those two additional fields are always carried over and that we leave our fingers off them.